src/main/java/org/jenkins/plugins/lockableresources/actions/ResourceVariableNameAction.java

```diff
@@ -0,0 +1,43 @@
+package org.jenkins.plugins.lockableresources.actions;
+
+import java.io.IOException;
+
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.model.EnvironmentContributor;
+import hudson.model.InvisibleAction;
+import hudson.model.Run;
+import hudson.model.StringParameterValue;
+import hudson.model.TaskListener;
+
+@Restricted(NoExternalUse.class)
+public class ResourceVariableNameAction extends InvisibleAction {
+
+	private final StringParameterValue resourceNameParameter;
```

Comment:
:ant: more straightforward (on both creator & consumer sides!) to split this into two `String` fields.